javaparser-core/src/main/java/com/github/javaparser/ast/NodeList.java

```diff
@@ -87,6 +87,27 @@ public boolean remove(Node node) {
         return innerList.remove(node);
     }
 
+    public boolean removeFirst() {
+        if (innerList.size() < 1) {
+            return false;
+        }
+        Node node = innerList.get(0);
+        notifyElementRemoved(0, node);
+        node.setParentNode(null);
+        return innerList.remove(0) != null;
+    }
```

Comment:
Isn't it easier to use the methods on NodeList instead of manipulating innerlist?

Reply:
You are absolutely right! I've simplified both removeFirst and removeLast so that they use NodeList's remove-method now.